src/xunit.performance.api/SafeTerminateHandler.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Microsoft.Xunit.Performance.Api.Native.Windows;
+using System;
+
+namespace Microsoft.Xunit.Performance.Api
+{
+    internal sealed class SafeTerminateHandler<T> : IDisposable
```

Comment:
I think this class deserves a comment block and some line-level comments to describe what it is for and what it does.  It certainly is a helpful wrapper class, but it also encapsulates some rather complex code, which we should document.